apps/backend/src/modules/groups/group.model.ts

```diff
@@ -0,0 +1,53 @@
+import { Model, type RelationMappings } from "objection";
+
+import {
+	AbstractModel,
+	DatabaseTableName,
+} from "~/libs/modules/database/database.js";
+import { PermissionModel } from "~/modules/groups/permission.model.js";
+import { UserModel } from "~/modules/users/user.model.js";
+
+class GroupModel extends AbstractModel {
+	public key!: string;
+
+	public name!: string;
+
+	public permissions!: { id: number }[];
+
+	public users!: { id: number }[];
+
+	public static override get relationMappings(): RelationMappings {
+		return {
+			permissions: {
+				join: {
+					from: `${DatabaseTableName.USER_GROUPS}.id`,
+					through: {
+						from: `${DatabaseTableName.USER_GROUPS_TO_PERMISSIONS}.userGroupId`,
+						to: `${DatabaseTableName.USER_GROUPS_TO_PERMISSIONS}.permissionId`,
+					},
+					to: `${DatabaseTableName.PERMISSIONS}.id`,
+				},
+				modelClass: PermissionModel,
+				relation: Model.ManyToManyRelation,
```

Comment:
When I have my electricity back, I'll check if I can put AbstractModel here (instead of Model)